group_test.go

```diff
@@ -213,3 +227,21 @@ func testConcurrentGroupWaitReallyWaits(t *testing.T, name string, g Executor) {
 		testingGroup.Wait()
 	})
 }
+
+func TestCanGoexit(t *testing.T) {
+	g := Unlimited(context.Background())
+	g.Go(func(context.Context) {
+		// Ideally we would test t.Fatal() here to show that parallel now plays
+		// nicely with the testing lib, but there doesn't seem to be any good
+		// way to xfail a golang test. As it happens t.Fatal() just sets a fail
+		// flag and then calls Goexit() anyway; if we treat nil recover() values
+		// as Goexit() (guaranteed since 1.21 with the advent of PanicNilError)
```

Comment:
_... realizes what it means that `panic(nil)` use to `recover()` as `nil`_

Reply:
there is some [deep lore](https://go-review.googlesource.com/c/sync/+/134395/3/errgroup/errgroup.go) to how `panic(nil)` used to work (and be worked around)